### PR TITLE
merge move 1.4

### PR DESCRIPTION
**Pull request by nanne007**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update or suggest changes to the docs at https://developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: crypto/crypto/src/hash.rs [hunk 1]
```diff
@@ -281,7 +301,7 @@ impl ser::Serialize for HashValue {
         S: ser::Serializer,
     {
         if serializer.is_human_readable() {
-            serializer.serialize_str(&self.to_hex())
+            serializer.serialize_str(&self.to_string())
```
Comment (jolestar) [inline crypto/crypto/src/hash.rs:304]: hash to_string 默认是短 hash 吧？

File: language/move-core/types/src/language_storage.rs [hunk 2]
```diff
@@ -120,8 +120,8 @@ impl Display for StructTag {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
         write!(
             f,
-            "0x{}::{}::{}",
-            self.address.short_str_lossless(),
+            "{}::{}::{}",
+            &self.address, // should not use short_str.
```
Comment (jolestar) [inline language/move-core/types/src/language_storage.rs:124]: 是不是我们也可以用 short_str 了?

**Review by jolestar — APPROVED** [review]

File: crypto/crypto/src/hash.rs (hunk 1 above)
Comment (jolestar) [reply crypto/crypto/src/hash.rs:304]: 哦，我们改了

File: language/move-core/types/src/vm_status.rs [hunk 3]
```diff
@@ -459,11 +459,25 @@ pub enum StatusCode {
...
+    // Config value doesn't exist
+    CONFIG_VALUE_DOES_NOT_EXIST = 28,
+    // Timestamp is invalid
+    INVALID_TIMESTAMP = 29,
+    // Coin to deposit is zero
+    COIN_DEPOSIT_IS_ZERO = 30,
+    // Token to destroy is not zero
+    DESTROY_TOKEN_NON_ZERO = 31,
+    // Block number mismatch
+    BLOCK_NUMBER_MISMATCH = 32,
```
Comment (jolestar) [inline language/move-core/types/src/vm_status.rs]: 我们自己增加的这些 error code 是不是可以废弃掉了。

Comment (nanne007) [reply language/move-core/types/src/vm_status.rs]: 改成开头了

**Pull request merged**